File: src/99beers.c
#include <stdio.h>

int main(int argc, char **argv)
{
    int beers = 99;

    while(1)
    {
        if(beers > 1)
        {
            printf("%d bottles of beer on the wall, %d bottles of beer.\n", beers, beers);
            beers--;
            if(beers > 1)
            {
                printf("Take one down, pass it around, %d bottles of beer on the wall.\n\n", beers);
            }
            else
            {
                printf("Take one down, pass it around, %d bottle of beer on the wall\n\n", beers);
            }
        }
        else
        {
            printf("%d bottle of beer on the wall, 1 bottle of beer\n", beers);
            printf("Take one down, pass it around, no more bottles of beer on the wall.\n\n");
            break;
        }
    }

    printf("No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n\n");

    return(0);
}

File: src/99beers.cs
namespace _99beers
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int initialBeerCount = 99;

            for (int i = initialBeerCount; i >= 0; i--)
            {
                Console.WriteLine(Beers(i));
            }
        }

        public static string Beers(int beerCount)
        {
            string lyrics;

            if (beerCount == 0)
            {
                lyrics = $"No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.";
            }
            else if (beerCount == 1)
            {
                lyrics = $"{beerCount} bottle of beer on the wall, {beerCount} bottle of bear.\nTake one down, and pass it around, no more bottles of beer on the wall.\n";
            }
            else
            {
                lyrics = $"{beerCount} bottles of beer on the wall, {beerCount} bottles of bear.\nTake one down, and pass it around, {beerCount - 1} bottles of beer on the wall.\n";
            }
            
            return lyrics;
        }
    }
}